### PR TITLE
Preserve timestamps for unchanged files.

### PR DESCRIPTION
**Pull request by coryan**
Adding `IncludeBlocks: Merge` apparently triggers a bug in
clang-format, where it changes the timestamp of the file even if
there are no formatting changes. I think we want that directive, so I
changed the check-style.sh script to workaround the issue.

While testing this I discovered that cmake-format has the same problem,
so I worked around that bug too.

Finally, I made some cosmetic changes to make my IDE happy inside this
script.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/2568)

### CODE REVIEW

**Comment by coryan** [comment]
@mr-salty you said LGTM, but looks like you accidentally forgot to click the approve button.

**Comment by devbww** [comment]
It looks like it might have already been fixed.

```
$ ls -l sign_blob_requests.cc
-rw-r--r-- 1 bww eng 2218 Apr  8 23:33 sign_blob_requests.cc
$ clang-format -i sign_blob_requests.cc
$ ls -l sign_blob_requests.cc
-rw-r--r-- 1 bww eng 2218 Apr  8 23:33 sign_blob_requests.cc
$ clang-format-7 -i sign_blob_requests.cc
$ ls -l sign_blob_requests.cc
-rw-r--r-- 1 bww eng 2218 Apr 27 23:27 sign_blob_requests.cc
$ clang-format -version
clang-format version google3-trunk (trunk r358772)
$ clang-format-7 -version  
clang-format version 7.0.1-6 (tags/RELEASE_701/final)
```

And given that we didn't notice it before, perhaps it was a regression in version 7.

File: ci/check-style.sh [hunk 1]
```diff
@@ -27,30 +27,52 @@ readonly BINDIR="$(dirname "$0")"
...
+
 # Apply cmake_format to all the CMake list files.
 #     https://github.com/cheshirekow/cmake_format
 find . \( "${ignore[@]}" \) -prune -o \
        \( -name 'CMakeLists.txt' -o -name '*.cmake' \) \
        -print0 |
-  xargs -0 cmake-format -i
+  while IFS= read -r -d $'\0' file; do
+    cmake-format "${file}" >"${file}.tmp"
+    replace_original_if_changed "${file}"
```
Comment (devbww) [inline ci/check-style.sh]: It's a little confusing that `"${file}.tmp"` isn't mentioned here.  I think I'd just make the function take two names (i.e., not assume `.tmp`).

**Comment by coryan** [comment]
PTAL.

File: ci/check-style.sh [hunk 2]
```diff
@@ -40,18 +40,18 @@ find google/cloud -name '*.h' -print0 |
...
 
-  local file="$1"
-  shift
+  local original="$1"
+  local reformatted="$2"
 
-  chmod --reference="${file}" "${file}.tmp"
-  if cmp -s "${file}" "${file}.tmp"; then
-    rm -f "${file}.tmp"
+  chmod --reference="${original}" "${reformatted}"
```
Comment (devbww) [inline ci/check-style.sh]: I'm not sure why I didn't originally put the "chmod" in the "mv" block, but we can do just that if we get back here.

**Pull request merged**